kuma/wiki/migrations/0003_remove_soft_document_deletion.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.20 on 2019-04-29 11:50
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+def purge_all_deleted_document(apps, schema_editor):
+    Document = apps.get_model('wiki', 'Document')
+    for document in Document.objects.filter(deleted=True):
+        document.delete()
```

Comment:
I think the `pre_delete` and `post_delete` signals are called manually when we do a soft-delete, so we should be able to ignore those.

However, I don't know the codebase enough to be confident about it.
I don't know, if it's expensive. If it's not, I would keep it that way to avoid any risk and if it is expensive, I would ask someone to double check to be sure.

I also believe that Kuma code merge policy enforces a minimum of 2 reviews from staff before merging on migration files, so if you and Ryan are both confident on this, I am happy to change it.

Reply:
This is a once-ever kinda thing. There are about 1,000 documents that this is going to affect. 
All of them have already been "manually" triggered to unindex and unpublish so we don't need to re-trigger the `pre_delete` signals that we'd lose.